trillian/ctfe/storage/storage.go

```diff
@@ -0,0 +1,17 @@
+package storage
+
+import (
+	"context"
+)
+
+// NonLeafCertificateChainStorage is an interface which allows CTFE binaries to use different storage implementations for non-leaf certificate chains.
+type NonLeafCertificateChainStorage interface {
```

Comment:
Maybe use a shorter name if possible? I was looking at the full branch, and some lines get very long sometimes. I can think of "IssuerChainStorage" [1] "IssuanceChainStorage" or "SubmissionChainStorage", what do you think? Feel free to disregard if you've already been through this and settled on this being the best possible name.

I'll admit [1] could me a bit missleading because some parts of ctgo mention the preIssuerChain.

Reply:
I do agree this name is a bit long. I thought about this but I couldn't find any better naming. Maybe @AlCutter or @mhutchinson have a better idea.

Reply:
`IssuanceChain` would be shorter and clearer.

Reply:
Renamed all `NonLeafCertificateChain` to `IssuanceChain`. Thanks for the suggestion.